### PR TITLE
feat: Implement BucketBlobStorage for BucketStorage

### PR DESCRIPTION
**Pull request by Xuanwo**
This PR will implement BucketBlobStorage for BucketStorage. And I will continue the work of `impl BlobStorage for BlobBucketDBStorage` in next PR.

Part of https://github.com/toeverything/OctoBase/issues/464

## Questions

`delete_blob` returns `Result<bool>`. However, opendal will always return success if file is not exist. Do we need to check file first before delete it?

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi @thorseraq, would you be willing to review for me? Thank you.

**Review by thorseraq — APPROVED** [review]

**Comment by thorseraq** [comment]
LGTM, Thanks for quick PR!

> delete_blob returns Result<bool>. However, opendal will always return success if file is not exist. Do we need to check file first before delete it?

Make sense, just realized there're some transaction semantic needed to ensure when combine db operation with s3, but it's not relevant with this PR, let's merge it

File: libs/jwst-storage/src/storage/blobs/bucket_local_db.rs [hunk 1]
```diff
@@ -67,23 +87,32 @@ impl BucketBlobStorage<JwstStorageError> for BucketStorage {
         hash: String,
         blob: Vec<u8>,
     ) -> JwstResult<String, JwstStorageError> {
-        todo!()
+        let workspace = self.get_workspace(workspace);
+        let key = self.build_key(workspace, hash);
+        let _ = self.op.write(&key, blob).await?;
+
+        // FIXME: what's the meaning of string?
+        Ok("".to_string())
```
Comment (thorseraq) [inline libs/jwst-storage/src/storage/blobs/bucket_local_db.rs:95]: It's useless, good to change method signature

File: libs/jwst-storage/src/storage/blobs/bucket_local_db.rs [hunk 2]
```diff
@@ -46,19 +46,39 @@ pub struct BucketStorage {
...
-#[allow(unused_variables)]
 #[async_trait]
 impl BucketBlobStorage<JwstStorageError> for BucketStorage {
     async fn get_blob(
         &self,
         workspace: Option<String>,
         id: String,
-        params: Option<HashMap<String, String>>,
+        // TODO: params is not used for now.
+        _params: Option<HashMap<String, String>>,
```
Comment (thorseraq) [inline libs/jwst-storage/src/storage/blobs/bucket_local_db.rs:75]: Useless, good to modify trait method signature

**Pull request merged**